server/datasources/requestDataSource.ts

```diff
@@ -1,9 +1,73 @@
 import CachedMongooseDataSource from './cachedMongooseDataSource'
 import { RequestCache } from '../database/cache'
 import { RequestDocument } from '../models/requestModel'
+import { ServerResponseInterface } from '../graphql/serverResponse'
+
+import { Types } from 'mongoose'
 
 export default class RequestDataSource extends CachedMongooseDataSource<RequestDocument> {
   constructor() {
     super(RequestCache)
   }
+  async createRequest(request: RequestDocument): Promise<ServerResponseInterface> {
```

Comment:
If we are putting these generic functions in cachedMongooseDataSource, please follow up with this and modify/remove these functions accordingly.

Reply:
Done